### ISSUE TITLE
Missing code context in traceback when the project is a package

### ISSUE THREAD

**Issue by reverbc** [bugsnag/bugsnag-python#245]
### Describe the bug
When the target project for Bugsnag is a python package instead of an application, setting the `project_root` will [trim the file path](https://github.com/bugsnag/bugsnag-python/blob/master/bugsnag/event.py#L196) so the file path of the exception traceback frame will become relative path (from `project_root` to the actual file). However, if the current working directory is not the `project_root` (which is usually true if the project is a python package), then the `bugsnag.Event._code_for` method will not be able to find the file to fetch the code context as the [linecache.getlines](https://github.com/bugsnag/bugsnag-python/blob/master/bugsnag/event.py#L218) is finding the file from cwd + relative file path, not `project_root` + relative file path.

### Environment
* Bugsnag version: 4.0.1
* Python version: 3.7.8
* Integration framework version: N/A

### Example code snippets

The folder structure for the example is:

```
consumer/
    entry.py
package/
    __init__.py
```

Content of `consumer/entry.py`:

```
import package

def main():
    package.entry()

if __name__ == "__main__":
    main()
```

Content of `package/__init__.py`:

```
import logging
import pathlib
import bugsnag
from bugsnag.handlers import BugsnagHandler

bugsnag.configure(
    api_key=<API Key>,
    project_root=str(pathlib.Path(__file__).resolve().parent),
)
logger = logging.getLogger()
handler = BugsnagHandler()
handler.setLevel(logging.ERROR)
logger.addHandler(handler)

def _boooom():
    return 123 + '123'

def entry():
    return _boooom()
```

Command to reproduce:

```
~/path/to/consumer $ PYTHONPATH=../ python entry.py
```

Expecting result:
![image](https://user-images.githubusercontent.com/783515/100295927-c3d6a000-2f3f-11eb-830b-02c123b18950.png)

Actual result:
![image](https://user-images.githubusercontent.com/783515/100295898-b7524780-2f3f-11eb-9f04-89eb4331f2bb.png)

**Comment by xander-jones** [comment]
Hey @reverbc, thanks for raising this and providing an example. I've reproduced, and work is already underway to fix this. Will let you know as soon as a release is available!

**Comment by kattrali** [comment]
Resolved in [v4.0.2](https://github.com/bugsnag/bugsnag-python/releases/tag/v4.0.2)
